src/librustc_mir/build/scope.rs

```diff
@@ -918,8 +891,11 @@ fn build_scope_drops<'tcx>(
     });
 
     // When we unwind from a drop, we start cleaning up from the next one, so
-    // we don't need this block.
-    unwind_blocks.next();
+    // we don't need this block. Unless we're in a generator, in which case we
+    // need this information.
+    if !is_generator {
```

Comment:
cc @Zoxc @matthewjasper 

The reason this `if` is here is because we would `drop(_4) -> [return: .., unwind: bbX]` and think that `_4` was StorageLive in `bbX`.

I could maybe just modify the `MaybeStorageLive` pass to consider drop statements so we're getting the same information. Any problems with doing that?

Reply:
cc @RalfJung; if a drop panics, can we assume the value is StorageDead?

Reply:
I don't think so. `drop` really is unrelated to the lifetime of the *enclosing* object, that's what `StorageDead` is for -- and I think @matthewjasper would know much more about how those are emitted than I do.